addon-resizer/nanny/kubernetes_client.go

```diff
@@ -53,6 +74,45 @@ func (k *kubernetesClient) CountNodes() (uint64, error) {
 	return uint64(len(result.Items)), err
 }
 
+func (k *kubernetesClient) countNodesThroughMetrics() (uint64, error) {
+	rawMetrics, err := k.clientset.Core().RESTClient().Get().RequestURI("/metrics").DoRaw()
```

Comment:
Two questions here:
Should we be concerned about the size of the response here, since we are reading all metrics?
Is the API server's metrics endpoint always accessible from the nodes?

Reply:
1. It may be larger for very small clusters, but it will be smaller for larger clusters (comparing to listing nodes). We still need to get exact numbers.

2. if they have permissions (similarly as for listing nodes or anything else) - then yes. Added comment.